play.pokemonshowdown.com/src/client-main.ts

```diff
@@ -1793,6 +1815,46 @@ export const PS = new class extends PSModel {
 		if (room) {
 			this.removeRoom(room);
 			this.update();
+			this.updateAutojoin();
+		}
+	}
+
+	updateAutojoin() {
+		if (!PS.server.registered) return;
+		if (!this.prefs.autojoin) this.prefs.autojoin = {};
+		if (!this.prefs.autojoin[this.server.id]) this.prefs.autojoin[this.server.id] = '';
+		let autojoins: string[] = [];
+		let autojoinCount = 0;
+		let rooms = this.rightRoomList;
+		for (let roomid of rooms) {
+			let room = PS.rooms[roomid];
+			if (!room) return;
+			if (room.type !== 'chat') continue;
```

Comment:
```suggestion
			if (room.type !== 'chat' || room.pmTarget) continue;
```

PM rooms are chatrooms now and need to be manually skipped.